language/stdlib/modules/AccountLimits.move

```diff
@@ -24,36 +22,55 @@ module AccountLimits {
     // them--any operations (sending/receiving/storing) that would cause us
     // to look at it will cause the transaction to abort.
     // All monentary amounts specified in microLBR
-    resource struct LimitsDefinition {
-        // The maximum inflow + outflow allowed during the specified time period.
-        max_total_flow: u64,
+    resource struct LimitsDefinition<CoinType> {
+        // The maximum inflow allowed during the specified time period.
+        max_inflow: u64,
+        // The maximum outflow allowed during the specified time period.
+        max_outflow: u64,
         // Time period, specified in microseconds
         time_period: u64,
         // The maximum that can be held
         max_holding: u64,
-        // Certification flag to say whether this limits definition is approved.
-        is_certified: bool,
     }
 
     // A struct holding account transaction information for the time window
     // starting at `window_start`.
-    resource struct Window {
+    resource struct Window<CoinType> {
         // Time window start in microseconds
         window_start: u64,
-        // The outflow + inflow during this time window
-        window_total_flow: u64,
+        // The inflow during this time window
+        window_inflow: u64,
+        // The inflow during this time window
+        window_outflow: u64,
         // The balance that this account has held during this time period.
         tracked_balance: u64,
         // address storing the LimitsDefinition resource that governs this window
-        limits_definition: address,
+        limit_address: address,
     }
 
+    const NOT_GENESIS: u64 = 0;
+    const INVALID_INITIALIZATION_ADDRESS: u64 = 1;
+    /// 24 hours in microseconds
+    const TIME_WINDOW: u64 = 86400000000;
```

Comment:
Nit: I think calling this `ONE_DAY` will make the code slightly more readable (I was looking at a function that uses this and thought it might be U64_MAX because of the name).